homeassistant/components/saj/sensor.py

```diff
@@ -40,24 +45,43 @@
     "": None,
 }
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({vol.Required(CONF_HOST): cv.string})
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_HOST): cv.string,
+        vol.Optional(CONF_TYPE, default=INVERTER_TYPES[0]): vol.In(INVERTER_TYPES),
+        vol.Optional(CONF_USERNAME, default=""): cv.string,
+        vol.Optional(CONF_PASSWORD, default=""): cv.string,
+    }
+)
 
 
 async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
     """Set up SAJ sensors."""
 
     remove_interval_update = None
+    wifi = config[CONF_TYPE] == INVERTER_TYPES[1]
 
     # Init all sensors
-    sensor_def = pysaj.Sensors()
+    sensor_def = pysaj.Sensors(wifi)
 
     # Use all sensors by default
     hass_sensors = []
 
     for sensor in sensor_def:
         hass_sensors.append(SAJsensor(sensor))
 
-    saj = pysaj.SAJ(config[CONF_HOST])
+    if not wifi:
+        saj = pysaj.SAJ(config[CONF_HOST])
+    else:
+        if config[CONF_USERNAME] != "" and config[CONF_PASSWORD] != "":
+            saj = pysaj.SAJ(
```

Comment:
We should check that credentials are ok. If not we should log error and return from platform setup.

Reply:
Thanks for the help, changed in the meantime!